doc/manual/merging.md

```diff
@@ -0,0 +1,117 @@
+# Merging records
+
+When you have a big configuration, the way to split it in several files
+(e.g.: separated by cathegorie or level of abstraction) is the merging operator.
+In nickel this is the `&` operator.
+
+For instence, having a server config and a firewall config, the network config
+could be:
+
+```text
+let
+server = import "server.ncl",
+firewall = import "firewall.ncl",
+in
+server & firewall
+```
+
+In the simple case where both records do not have common fields, it gives a
+record which is the union of both.
+
+If you have to merge records with fields in commons, you will use two new concepts:
+
+- default annotation,
+- overwriting,
```

Comment:
Are these two concepts or just one?

There are other merging cases:
- merging two record fields.
- merging a field against a field which merely has a contract attached (this emulate mix-ins in particular).